### PR TITLE
Add tests

### LINKED ISSUE #6: Write good tests for the module
After #5 is implemented, this should probably become better. (This would probably also need the option to pass terminal width as an argument, because travis terminal widths evaluate to 0, check [sample travis builds](https://travis-ci.org/icyflame/bar-horizontal/builds/297810567#L1060) )

### PR DESCRIPTION
**Pull request by jrmykolyn**
Completed a first pass at a new set of tests for the `barHorizontal()` module. Fixes #6.

NOTE(S):
- Tests based on comments outlined in: https://github.com/icyflame/bar-horizontal/issues/6
- Tests written using existing mocha framework.
- New test suite lives in `test/` dir.
- New tests suit contains stubs/placeholder implementations for 'warnings' functionality. To be revisited.
- No adjustments made to existing 'test.js' file.

### CODE REVIEW

File: test/bar-horizontal.spec.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    let stdout = capcon.captureStdout( function() {
+      return barHorizontal( { a: 1, b: 2, c: 3 }, { labels: true } );
+    } );
+
+    assert.ok( stdout.indexOf( 'a' ) !== -1 );
+  } );
+
+
+  xit( 'Should not display warnings by default.', function() {
+    /// TODO: Unclear how to test for this, as warnings may not be triggered even when enabled.
```
Comment (icyflame) [inline test/bar-horizontal.spec.js]: You can use the `width` option. For eg, `barHorizontal([100, 10], { width: 4, warnings: true })` will always raise a warning. (Please rebase onto latest code, I had goofed up the check for warnings in my last release)

File: index.js [hunk 2]
```diff
@@ -2,6 +2,16 @@
 var _ = require('lodash');
 
 module.exports = function (inputObj, opts) {
+  // Throw error if `inputObj` does not exist or is not an array/object.
+  if ( !inputObj || typeof inputObj !== 'object'  ) {
+    throw new Error( 'Whoops! Please ensure that `barHorizontal()` is invoked with either an array or object.' );
+  }
+
+  // Throw error if `inputObj` is an empty array or key-less object.
+  if ( ( typeof inputObj.length !== 'undefined' && !inputObj.length ) || ( typeof inputObj.length === 'undefined' && !Object.keys( inputObj ).length ) ) {
```
Comment (icyflame) [inline index.js]: Is this `typeof` check required? Now that we know `inputObj` is definitely an object, wouldn't it be enough to ensure that the length isn't 0?

File: index.js [hunk 3]
```diff
@@ -2,6 +2,16 @@
 var _ = require('lodash');
 
 module.exports = function (inputObj, opts) {
+  // Throw error if `inputObj` does not exist or is not an array/object.
+  if ( !inputObj || typeof inputObj !== 'object'  ) {
```
Comment (icyflame) [inline index.js]: This code looks odd, better to just check undefined. We need an object, anything else will break the module anyway. So, just the `typeof` check should be enough. Let me know if there is some other case you had in mind when you wrote this. :slightly_smiling_face:

Comment (jrmykolyn) [reply index.js]: I was thinking that the module would error out immediately if/when invoked with an invalid value (eg. anything other than an array of object). 

- `typeof inputObj === 'undefined` would ensure that an argument is present, but not that it's of the correct type.
- `typeof inputObj === 'object` would ensure that the argument is either an object or array, but a `null` value would also be let through.
- The leading `!inputObj` was added to protect against `null` values.

Maybe there's a more elegant solution, or maybe it's too strict? I'm very happy to make any changes you request, so let me know what you think>

File: index.js (hunk 2 above)
Comment (jrmykolyn) [reply index.js]: I have mixed feelings about this block as well.

I was thinking that at this point, we don't actually know if the `inputObj` is an object or an array, and object literals don't expose a `length` property. Maybe the whole condition is not required (eg. DON'T error if the `inputObj` is empty/keyless)?

File: test/bar-horizontal.spec.js (hunk 1 above)
Comment (jrmykolyn) [reply test/bar-horizontal.spec.js]: Awesome. Will revisit and update, thanks.

File: index.js (hunk 2 above)
Comment (icyflame) [reply index.js]: What do you think about:

```js
typeof inputObj === 'object' && inputObj != null && Object.keys(inputObj).length > 0
```

(Arrays are just json objects) I think this should be [enough](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof#Examples)

And the error message could be `Invoke bar-horizontal with a non-empty array or object`. This single check would replace both your `if` statements.

P.S lodash's `isEmpty` also does all this checking. But it is overkill in this case.

Comment (jrmykolyn) [reply index.js]: Awesome, thanks for the direction. Will update.

File: index.js (hunk 3 above)
Comment (jrmykolyn) [reply index.js]: UPDATE: This is addressed by your proposed update in the comment(s) below.

**Comment by jrmykolyn** [comment]
@icyflame All requested changes have been implemented. Let me know what you think!

**Comment by icyflame** [comment]
@jrmykolyn Thanks! The changes are fine on my computer. 

As you can see though, Travis isn't working because it is running on a width 0 terminal. Maybe we should use a fixed width of 200 in all these tests to ensure that we get the same output no matter what terminal we run on. (The current tests will not work if the terminal I am running the tests on is very small!) :slightly_smiling_face: Very minor change, I will fix it tomorrow morning.

**Comment by jrmykolyn** [comment]
@icyflame Thanks for approving!

Apologies about the TravisCI integration, I definitely overlooked that. Are you going to take care of the Travis-related adjustments?

**Comment by icyflame** [comment]
Hey @jrmykolyn, I have merged this in now. I will create a release soon. I am thinking of some more tests that I would like to have. Thanks for your time! :+1: 

Meanwhile, I added a `noPrint` option to tell the module to not print anything to the console. This makes the test output look [prettier](https://travis-ci.org/icyflame/bar-horizontal/builds/304612934#L539) and easier to read. Have a look at the latest master, tell me if you see anything amiss. :smile:

**Comment by jrmykolyn** [comment]
@icyflame 

Awesome, thanks for merging that in. Also, test output looks great in Travis!

Let me know if I can help with anything else.

**Pull request merged**